### PR TITLE
Adds menu on folder to change branch

### PR DESCRIPTION
**Pull request by marcelogdeandrade**
I implemented what was requested on issue #501 .

Screenshot

I continued using [Ggit](https://valadoc.org/ggit-1.0/Ggit.Index.html) lib to enumerate branches and to set a new head.

### CODE REVIEW

File: src/FolderManager/ProjectFolderItem.vala [hunk 1]
```diff
@@ -74,6 +74,33 @@ namespace Scratch.FolderManager {
...
         }
 
+        public void change_branch (string ref_name) {
+            bool success = git_repo.set_head (ref_name);
+        }
+
+        public Gtk.MenuItem create_change_branch_menu () {
+            var change_branch_menu = new Gtk.Menu ();
+
+            var cur_branch = git_repo.get_head () as Ggit.Branch;
```
Comment (davidmhewitt) [inline src/FolderManager/ProjectFolderItem.vala]: Please use the `is_branch` method on the `Ref` returned from `get_head` before casting it to a `Ggit.Branch`. It's possible it's not a branch.

File: src/FolderManager/ProjectFolderItem.vala [hunk 2]
```diff
@@ -74,6 +74,33 @@ namespace Scratch.FolderManager {
             }
         }
 
+        public void change_branch (string ref_name) {
+            bool success = git_repo.set_head (ref_name);
```
Comment (davidmhewitt) [inline src/FolderManager/ProjectFolderItem.vala]: No need for this bool if it's not used anywhere.

File: src/FolderManager/ProjectFolderItem.vala [hunk 3]
```diff
@@ -74,6 +74,33 @@ namespace Scratch.FolderManager {
...
+        }
+
+        public Gtk.MenuItem create_change_branch_menu () {
+            var change_branch_menu = new Gtk.Menu ();
+
+            var cur_branch = git_repo.get_head () as Ggit.Branch;
+            var branches = git_repo.enumerate_branches (Ggit.BranchType.LOCAL);
+
+            foreach (var ref_branch in branches) {
+                Ggit.Branch branch = ref_branch as Ggit.Branch;
```
Comment (davidmhewitt) [inline src/FolderManager/ProjectFolderItem.vala]: You can use `var` here as the type should be inferred from `as Ggit.Branch`

File: src/FolderManager/ProjectFolderItem.vala [hunk 4]
```diff
@@ -74,6 +74,33 @@ namespace Scratch.FolderManager {
...
+                string branch_name = branch.get_name ();
```
Comment (davidmhewitt) [inline src/FolderManager/ProjectFolderItem.vala]: Use `var` here too

File: src/FolderManager/ProjectFolderItem.vala [hunk 5]
```diff
@@ -74,6 +74,33 @@ namespace Scratch.FolderManager {
...
+                if (branch_name != cur_branch.get_name ()) {
+                    string ref_name = ref_branch.get_name ();
```
Comment (davidmhewitt) [inline src/FolderManager/ProjectFolderItem.vala]: `var` here too

**Comment by marcelogdeandrade** [comment]
@davidmhewitt Thanks for the review! I did the requested changes.

When checking if git_repo.get_head() is a branch, I thought that it would be better to do when adding the submenu on the main menu. If I did it on my `create_change_branch_menu` function, an empty submenu would be added.

**Comment by snowparrot** [comment]
ping.

File: src/FolderManager/ProjectFolderItem.vala [hunk 6]
```diff
@@ -88,6 +115,7 @@ namespace Scratch.FolderManager {
             menu.append (close_item);
             menu.append (create_submenu_for_new ());
             menu.append (delete_item);
+            if (git_repo != null && git_repo.get_head ().is_branch ()) { menu.append (create_change_branch_menu ()); }
```
Comment (jeremypw) [inline src/FolderManager/ProjectFolderItem.vala]: One line if clauses are contrary to current elementary code style. Always use format:

```
if (condition) {
    code;
}
```

in new code.

**Comment by marcelogdeandrade** [comment]
Oh, it's been a couple of months since I opened this PR, maybe I removed my fork by accident, I didn't see this open PR.

**Comment by jeremypw** [comment]
Will you be able to resurrect it? It looks like a useful thing.

**Comment by jeremypw** [comment]
Closing as original repository appears to have been lost.

**Pull request closed without merging**